File: surplus-hub-react/packages/core/src/types/community.ts
import { PaginationMeta } from "./material";

export interface CommunityPost {
  id: string;
  title: string;
  content: string;
  category: string;
  imageUrl?: string;
  authorId: string;
  authorName: string;
  views: number;
  likesCount: number;
  createdAt: string;
}

export interface CommunityPostsResponse {
  data: CommunityPost[];
  meta?: PaginationMeta;
}


export interface CommunityPostCreate {
  title: string;
  content: string;
  category: string;
  imageUrl?: string;
}

export interface CommunityPostsQueryParams {
  page?: number;
  limit?: number;
  category?: string;
  authorId?: string;
}

File: surplus-hub-api-v3/app/models/community.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base

class Post(Base):
    __tablename__ = "posts"

    id = Column(Integer, primary_key=True, index=True)
    author_id = Column(Integer, ForeignKey("users.id"))
    
    title = Column(String, nullable=False)
    content = Column(Text, nullable=False)
    category = Column(String, index=True) # QnA, KnowHow, Safety, Info
    
    views = Column(Integer, default=0)
    likes_count = Column(Integer, default=0)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    author = relationship("User", backref="posts")
    
    # Simple image handling for posts (single or multiple, usually handled by a separate table or array, keeping simple for now)
    image_url = Column(String, nullable=True) 

class Comment(Base):
    __tablename__ = "comments"
    
    id = Column(Integer, primary_key=True, index=True)
    post_id = Column(Integer, ForeignKey("posts.id"))
    author_id = Column(Integer, ForeignKey("users.id"))
    
    content = Column(Text, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    post = relationship("Post", backref="comments")
    author = relationship("User")
